### PR TITLE
Reimplement extraction using SexpProcessor.

### PR DESCRIPTION
**Pull request by martinpovolny**
Replacement of the ruby_parser hack by use of sexp_processor. See https://github.com/seattlerb/ruby_parser/issues/198 for details.

### CODE REVIEW

**Comment by martinpovolny** [comment]
@mzazrivec: can you, please help test this?

**Comment by mzazrivec** [comment]
@martinpovolny 

```
$ rake gettext:find
...
#<GetText::RubyLexX:0x0055bb5f822588>#<GetText::RubyLexX:0x0055bb5f822588>Error parsing app/views/ops/_db_info.html.haml
rake aborted!
NameError: uninitialized constant RubyGettextExtractor::SexpProcessor
```

**Comment by mzazrivec** [comment]
@martinpovolny With the `sexp_processor` gem added into Gemfile:

```
#<GetText::RubyLexX:0x0055af91f8be20>#<GetText::RubyLexX:0x0055af91f8be20>Error parsing app/views/ops/_db_info.html.haml
rake aborted!
ArgumentError: wrong number of arguments (3 for 1..2)
/home/milan/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/ruby_gettext_extractor.rb:11:in `parse_string'
/home/milan/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/base_parser.rb:12:in `parse'
```

**Comment by martinpovolny** [comment]
@grosser: I think that this PR is ready for review by you. Please? ;-)

The failure in this PR here: https://travis-ci.org/grosser/gettext_i18n_rails/jobs/83915877 is most likely caused by usage of too old ruby_parser gem that does not support ruby 2.2. Shall I add the newer ruby_parser gem as a dependancy to gettext_i18n_rails?

@mzazrivec : all issues should be fixed

**Comment by grosser** [comment]
yeah, bump whatever is needed to get us green :)

**Comment by martinpovolny** [comment]
@grosser : done

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@
 require 'fast_gettext'
 require 'gettext_i18n_rails'
 require 'temple'
+require 'sexp_processor'
```
Comment (grosser) [inline spec/spec_helper.rb]: that require should be in the extractor ?

File: gettext_i18n_rails.gemspec [hunk 2]
```diff
@@ -15,7 +15,8 @@ Gem::Specification.new name, GettextI18nRails::VERSION do |s|
   s.add_development_dependency "haml"
   s.add_development_dependency "rake"
   s.add_development_dependency "rails"
-  s.add_development_dependency "ruby_parser", ">= 3"
+  s.add_development_dependency "ruby_parser", ">= 3.7.1"
```
Comment (grosser) [inline gettext_i18n_rails.gemspec]: can you put a `gem "ruby_parser", ">= 3.7.1"` where it is required in the codebase so nobody runs into weirdness ?

Comment (martinpovolny) [inline gettext_i18n_rails.gemspec]: I don't quite understand what you mean? Shall I add the 'gem' call to the file that needs the parser? Meaning into `lib/gettext_i18n_rails/ruby_gettext_extractor.rb`?

The version is needed to support ruby 2.2. Older ruby works with older versions of the gem.

Please, try to reformulate what you want me to do. I'll do whatever is needed for sure ;-)

Comment (grosser) [inline gettext_i18n_rails.gemspec]: yes that's what I meant :)

should be fine to force old users to upgrade ruby_parser ...

**Comment by grosser** [comment]
can you move the require into lib and add the `gem` calls for ruby_parser + sexp_processor ?

**Comment by martinpovolny** [comment]
@grosser : moved the require but I am still confused, about the other change. Would you, please, make patch with that change for me?

**Comment by grosser** [comment]
add them here https://github.com/grosser/gettext_i18n_rails/commit/c2a081284c72d8b2f9149764c075a83fe6c34cf7

released in 1.3.0

**Comment by martinpovolny** [comment]
Ah, sorry, I did not notice the Gemfile.lock was part of the project. Thx!

**Pull request merged**